web/themes/new_weather_theme/assets/js/components/alertMap.js

```diff
@@ -0,0 +1,118 @@
+  const alertId = alert.dataset.alertId;
+
+  const lat = Number.parseFloat(alert.dataset.lat);
+  const lon = Number.parseFloat(alert.dataset.lon);
+
+  const map = L.map(`wx_alert_map_${alertId}`).setView([lat, lon], 8);
+
+  // Leaflet is managed by a Ukrainian team. The default attribution they put on
+  // maps includes a Ukrainian flag to show their national pride. But as an
+  // official website of the US Government, that might not be appropriate for
+  // us, so turn off the attribution. We're not likely to use Leaflet in the
+  // end anyway, but if we do, we'll figure out how to put back attributions
+  // without the flag then.
+  map.attributionControl.setPrefix("");
+
+  L.esri.Vector.vectorBasemapLayer("arcgis/streets", {
+    apiKey:
+      "AAPK1dd93729edc54e84ade1ea5dc0f4f9d3EPexfd5qirlO3QtHGBj5JQL7iUYHQOb4yLjfKEYFLcyN9PlMd87lMjjv8D3DxDsQ",
+  }).addTo(map);
+
+  L.geoJSON(geoJSON, { style: { color: "#F00", opacity: 0.6 } }).addTo(map);
+
+  const locationIcon = L.divIcon({
+    className: "weathergov-location-marker",
+  });
+  L.marker([lat, lon], { icon: locationIcon, interactive: false }).addTo(map);
+
+  // Hide the location marker from screen readers and remove it from the tab
+  // order. It's not interactive, so there's no reason it should be focusable.
+  const locationMarker = document.querySelector(
+    `#wx_alert_map_${alertId} .weathergov-location-marker`,
+  );
+  if (locationMarker) {
+    locationMarker.setAttribute("aria-hidden", "true");
+    locationMarker.setAttribute("tabindex", "-1");
+  }
+};
+
+const waitForAlertAccordions = () => {
+  const alerts = document.querySelectorAll("wx-alert-map");
+
+  for (const alert of alerts) {
+    // We only want to show an alert map if the alert's accordion is open. If it
+    // is currently closed, we'll wait for it to open and then display it. We
+    // will only do this the first time the accordion is opened, so we don't
+    // have to worry about maps being added multiple times.
+
+    // Traverse up the DOM tree to the nearest node that matches this selector.
+    // This is our container.
+    const parent = alert.closest("weathergov-alerts > div");
+    const button = parent.querySelector("button");
+    const expanded = button.getAttribute("aria-expanded") === "true";
+
+    if (expanded) {
+      setupMap(alert);
+    } else {
+      // If we're not currently expanded, setup a mutation observer. Since we
+      // don't wrap the accordions, we
+      const mutationObserver = new MutationObserver((_, observer) => {
+        const expandedNow =
+          alert
+            .closest("weathergov-alerts > div")
+            .querySelector("button")
+            .getAttribute("aria-expanded") === "true";
+
+        if (expandedNow) {
+          setupMap(alert);
+          observer.disconnect();
+        }
+      });
+      mutationObserver.observe(button, { attributes: true });
+    }
+  }
+};
```

Comment:
Would be nice if these accordions dispatched some kind of custom event. Anyway, to my reading this seems functionally equivalent to adding a click handler to the accordion, doing all the stuff, and then removing the handler when it's done the first time. Unless, of course, something besides a click opens the toggle...